test-install.sh

```
@@ -0,0 +1,309 @@
+
+uname() {
+    case "$1" in
+        -s) echo "$mock_uname_s" ;;
+        -m) echo "$mock_uname_m" ;;
+        *) echo "unknown" ;;
+    esac
+}
+
+# Platform detection function (copied from install.sh)
+detect_platform() {
+    local os=$(uname -s | tr '[:upper:]' '[:lower:]')
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)